### PR TITLE
The category of groupoids

### PR DESCRIPTION
**Pull request by sstucki**
Another slice of #40: the category of groupoids.

This is the "weak" version of said category, using natural iso as functor equality, just as `Categories.Category.Instance.Cats`. A strict version (akin to `StrictCats`) will follow in a separate PR.

~~The reason for also including the change to `Categories.Morphism.IsoEquiv` in this PR is that the proof of `F-resp-⁻¹` uses `to-unique`.~~ (This is now in #51.)

### CODE REVIEW

**Comment by sstucki** [comment]
This PR has been open for a while. Is there anything problematic about it?

If so, please let me know how I can improve it.

**Comment by JacquesCarette** [comment]
It came in when I was super-busy, and I haven't been systematic enough to go back to look at it. Thanks for the ping, I'll look at it in detail now.

File: Categories/Morphism/IsoEquiv.agda [hunk 1]
```diff
@@ -68,3 +69,59 @@ Isos = record
       }
   }
   where open Equiv
+
+----------------------------------------------------------------------
+
+-- An alternative notion of equality on isomorphisms that only
```
Comment (JacquesCarette) [inline Categories/Morphism/IsoEquiv.agda]: Could you please split this into its own PR?

File: Categories/Category/Instance/Groupoids.agda [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import Categories.Morphism.IsoEquiv as IsoEquiv
+open import Categories.NaturalTransformation.NaturalIsomorphism
+  using (NaturalIsomorphism; associator; unitorˡ; unitorʳ; isEquivalence; _ⓘₕ_)
+private
+  variable
+    o ℓ e : Level
+
+-- A groupoid is a category that has a groupoid structure
+--
+-- This is not the same as Categories.Category.Groupoid.Groupoid.
```
Comment (JacquesCarette) [inline Categories/Category/Instance/Groupoids.agda]: I agree that the current `Categories.Category.Groupoid.Groupoid` is ill-named and should be `IsGroupoid`. That should be done - and this local definition removed.

I would prefer if things were done in that order though.  But if you're ok with doing both, I'm also ok with pulling this in first and trust that it will shortly be refactored.

File: Categories/Morphism/IsoEquiv.agda (hunk 1 above)
Comment (sstucki) [reply Categories/Morphism/IsoEquiv.agda]: Done in #51.

File: Categories/Category/Instance/Groupoids.agda (hunk 2 above)
Comment (sstucki) [reply Categories/Category/Instance/Groupoids.agda]: Done in #55.

**Comment by sstucki** [comment]
I refactored the PR further, as requested by @JacquesCarette.

It now depends on #51 and #55 (see above for details).

File: Categories/Category/Instance/Groupoids.agda [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import Categories.Morphism.IsoEquiv as IsoEquiv
+open import Categories.NaturalTransformation.NaturalIsomorphism
+  using (NaturalIsomorphism; associator; unitorˡ; unitorʳ; isEquivalence; _ⓘₕ_)
+private
+  variable
+    o ℓ e : Level
+
+-- A groupoid is a category that has a groupoid structure
+
+record Groupoid (o ℓ e : Level) : Set (suc (o ⊔ ℓ ⊔ e)) where
```
Comment (JacquesCarette) [inline Categories/Category/Instance/Groupoids.agda]: This should still be split off and put into `Categories.Category.Groupoid`.

Comment (sstucki) [reply Categories/Category/Instance/Groupoids.agda]: Hm. Should we then put `IsGroupoid` in that same module, or do you prefer to have both `Categories.Category.Groupoid` and `Categories.Category.IsGroupoid` as separate modules?

Comment (JacquesCarette) [reply Categories/Category/Instance/Groupoids.agda]: I have no strong opinions there.  I know the stdlib splits them, but for now, I would be fine with putting them all in the same module.

Comment (sstucki) [reply Categories/Category/Instance/Groupoids.agda]: OK, I put them in the same file. Less clutter.

**Comment by sstucki** [comment]
In the latest version, `Groupid` and `IsGroupoid` are now both in `Categories.Category.Groupoid`, while the actual category of groupoids is in `Categories.Category.Instance.Groupoids`.

The reason so many files are changed is that the `Categories.Category.IsGroupoid` module disappeared.
